app/middleware/auth.js

```diff
@@ -8,12 +8,13 @@ import jwt from 'jsonwebtoken';
 
 const auth = (req, res, next) => {
   let token = req.headers.authorization || req.headers['x-access-token'];
-  token = token.replace('Bearer ', '');
 
   if (!token) {
     return res.status(401).json({ message: 'Unauthorized Access' });
   }
 
+  token = token.replace('Bearer ', '');
```

Comment:
Why'd you change the position of this?

Reply:
I removed it because it is the reason the Game API is returning 404. For the unauthenticated user, the token will be undefined and the replace function will throw an error. Currently, if the token is undefined, this block of code "if (!token) {
     return res.status(401).json({ message: 'Unauthorized Access' });
   }" will be executed. therefore undefined token won't reach replace method